src/cmd/services/m3query/config/config.go

```diff
@@ -177,6 +183,14 @@ type Configuration struct {
 	Debug config.DebugConfiguration `yaml:"debug"`
 }
 
+// WriteWorkerPoolOrDefault returns the write worker pool config or default.
+func (c Configuration) WriteWorkerPoolOrDefault() xconfig.WorkerPoolPolicy {
```

Comment:
nit: may be worth renaming these to `WriteWorkerPoolPolicyOrDefault`? happy to punt though